worlds/lufia2ac/__init__.py

```diff
@@ -0,0 +1,351 @@
+    The Ancient Cave is a roguelike dungeon crawling game built into
+    the RGP Lufia II. Face 99 floors of ever harder to beat monsters,
+    random items and find new companions on the way to face the Royal
+    Jelly in the end. Can you beat it?
+    """
+    game: str = "Lufia II Ancient Cave"
+    web = L2ACWeb()
+
+    option_definitions: Dict[str, Option[Any]] = l2ac_option_definitions
+    item_name_to_id: Dict[str, int] = l2ac_item_name_to_id
+    location_name_to_id: Dict[str, int] = l2ac_location_name_to_id
+    item_name_groups: Dict[str, Set[str]] = {
+        "Blue chest items": {name for name, data in l2ac_item_table.items() if data.type is ItemType.BLUE_CHEST},
+        "Capsule monsters": {name for name, data in l2ac_item_table.items() if data.type is ItemType.CAPSULE_MONSTER},
+        "Party members": {name for name, data in l2ac_item_table.items() if data.type is ItemType.PARTY_MEMBER},
+    }
+    data_version: int = 1
+    required_client_version: Tuple[int, int, int] = (0, 1, 6)
+    required_server_version: Tuple[int, int, int] = (0, 2, 4)
+
+    # L2ACWorld specific properties
+    rom_name: Optional[bytearray]
+
+    blue_chest_chance: Optional[int]
+    blue_chest_count: Optional[int]
+    boss: Optional[Boss]
+    capsule_cravings_jp_style: Optional[int]
+    capsule_monsters_available: Optional[int]
+    capsule_starting_form: Optional[CapsuleStartingForm]
+    capsule_starting_level: Optional[CapsuleStartingLevel]
+    crowded_floor_chance: Optional[int]
+    death_link: Optional[int]
+    final_floor: Optional[int]
+    gear_variety_after_b9: Optional[int]
+    goal: Optional[int]
+    healing_floor_chance: Optional[int]
+    initial_floor: Optional[int]
+    iris_floor_chance: Optional[int]
+    iris_treasures_required: Optional[int]
+    master_hp: Optional[int]
+    party_members_available: Optional[int]
+    party_starting_level: Optional[PartyStartingLevel]
+    run_speed: Optional[int]
+    starting_capsule: Optional[int]
+    starting_party: Optional[StartingParty]
+
+    def __init__(self, multiworld: MultiWorld, player: int):
+        super().__init__(multiworld, player)
+
+    @classmethod
+    def stage_assert_generate(cls, _multiworld: MultiWorld) -> None:
+        rom_file: str = get_base_rom_path()
+        if not os.path.exists(rom_file):
+            raise FileNotFoundError(f"Could not find base ROM for {cls.game}: {rom_file}")
+
+        # # uncomment this section to recreate the basepatch
+        # # (you will need to provide "asar-x64.dll" and "asar.py" in the basepatch directory)
+        # from .basepatch import create_basepatch
+        # create_basepatch()
+
+    def generate_early(self) -> None:
+        self.rom_name = \
+            bytearray(f"L2AC{__version__.replace('.', '')[:3]}_{self.player}_{self.multiworld.seed}", "utf8")[:21]
+        self.rom_name.extend([0] * (21 - len(self.rom_name)))
+
+        self.blue_chest_chance = self.multiworld.blue_chest_chance[self.player].value
```

Comment:
i would recommend creating a lookup dictionary if you want to go this route instead of hardcoding every single option like this. There's examples in overcooked 2 and zillion, as well as #993